pandas/tests/io/test_sql.py

```diff
@@ -4355,3 +4355,19 @@ def test_xsqlite_if_exists(sqlite_buildin):
         (5, "E"),
     ]
     drop_table(table_name, sqlite_buildin)
+
+
+@pytest.mark.parametrize("dtype_backend", ["pyarrow", "numpy_nullable", lib.no_default])
```

Comment:
also found that this was failing for `dtype_backend=numpy_nullable` so implemented a fix and added test cases for each possible `dtype_backend`